src/main/kotlin/org/rust/lang/core/macros/MacroExpansionTask.kt

```diff
@@ -87,6 +88,15 @@ class MacroExpansionTask(
 
     private fun updateMacrosFiles(allDefMaps: List<CrateDefMap>) {
         val contentRoot = "/$MACRO_EXPANSION_VFS_ROOT/$projectDirectoryName"
+
+        if (lastUpdatedMacrosAt.isEmpty()) {
```

Comment:
Is it ok that refresh can happen several times if `allDefMaps` are empty for some reason?

Reply:
Ok, I've changed it to `if (lastUpdatedMacrosAt.isEmpty() && allDefMaps.isNotEmpty())`, in this case it seems impossible to execute it twice. And in the case when `allDefMaps.isEmpty()` there isn't a macro expansion anyway